game-core/src/main/java/games/strategy/triplea/attachments/PoliticalActionAttachment.java

```diff
@@ -146,4 +150,22 @@ public void validate(final GameData data) throws GameParseException {
       throw new GameParseException("value: relationshipChange can't be empty" + thisErrorMsg());
     }
   }
+
+  @Override
+  protected Map<String, AttachmentProperty<?>> createPropertyMap() {
+    return ImmutableMap.<String, AttachmentProperty<?>>builder()
+        .put("relationshipChange",
+            AttachmentProperty.of(
+                this::setRelationshipChange,
+                this::setRelationshipChange,
+                this::getRelationshipChange,
+                this::resetRelationshipChange))
+        .putAll(super.createPropertyMap())
```

Comment:
Can we try to refactor this to use composition?

The call-chain to A->B->C, going up to parent classes is a bit complex. We can instead reference each one we want to include explicitly. Let's say for example we needed a new abstract class somewhere in the middle of the chain, we could do so without impacting any child dependency.

Reply:
@DanVanAtta
I actually do think this inheritance from the super class is good style, however the putAll statement should probably always be at the beginning, so in case of duplicate keys the child class overrides them.